User.php

```diff
@@ -0,0 +1,579 @@
+
+		db_update('site_users',$id,array('user'=>$user['new_user'],'pass'=>$pass1,'no_logins'=>'Y'));
+		
+		$email1 = SiteEmail::getRecord('forgot');
+		Email::send($CFG->form_email,$email,$email1['title'],$CFG->form_email_from,false,$email1['content'],$user);
+	}
+	
+	
+	
+	function registerNew($info) {
+		global $CFG;
+		
+		if (!is_array($info))
+			return false;
+		
+		$info['email'] = preg_replace("/[^0-9a-zA-Z@\.\!#\$%\&\*+_\~\?\-]/", "",$info['email']);
+		$exist_id = self::userExists($info['email']);
+		if ($exist_id > 0) {
+			$user_info = DB::getRecord('site_users',$exist_id,0,1);
+			$email = SiteEmail::getRecord('register-existing');
+			Email::send($CFG->form_email,$info['email'],$email['title'],$CFG->form_email_from,false,$email['content'],$user_info);
+			return false;
+		}
+
+		$new_id = self::getNewId();
+		if ($new_id > 0) {
+			$sql = 'SELECT id FROM fee_schedule ORDER BY from_usd ASC LIMIT 0,1';
+			$result = db_query_array($sql);
+			
+			$pass1 = self::randomPassword(12);
+			$info['first_name'] = preg_replace("/[^\da-z ]/i", "",$info['first_name']);
+			$info['last_name'] = preg_replace("/[^\da-z ]/i", "",$info['last_name']);
+			$info['country'] = preg_replace("/[^0-9]/", "",$info['country']);
+			$info['user'] = $new_id;
+			$info['pass'] = Encryption::hash($pass1);
+			$info['date'] = date('Y-m-d H:i:s');
+			$info['confirm_withdrawal_email_btc'] = 'Y';
+			$info['confirm_withdrawal_email_bank'] = 'Y';
+			$info['notify_deposit_btc'] = 'Y';
+			$info['notify_deposit_bank'] = 'Y';
+			$info['notify_login'] = 'Y';
+			$info['no_logins'] = 'Y';
+			$info['fee_schedule'] = $result[0]['id'];
+			$info['default_currency'] = preg_replace("/[^0-9]/", "",$info['default_currency']);
+            $info['deactivated'] = 'N';
+			unset($info['terms']);
+			
+			$record_id = db_insert('site_users',$info);
+		
+			require_once('../lib/easybitcoin.php');
+			$bitcoin = new Bitcoin($CFG->bitcoin_username,$CFG->bitcoin_passphrase,$CFG->bitcoin_host,$CFG->bitcoin_port,$CFG->bitcoin_protocol);
+			$new_address = $bitcoin->getnewaddress($CFG->bitcoin_accountname);
+			db_insert('bitcoin_addresses',array('address'=>$new_address,'site_user'=>$record_id,'date'=>date('Y-m-d H:i:s')));
+		
+			$info['pass'] = $pass1;
+			$email = SiteEmail::getRecord('register');
+			Email::send($CFG->form_email,$info['email'],$email['title'],$CFG->form_email_from,false,$email['content'],$info);
+		
+			$email = SiteEmail::getRecord('register-notify');
+			$info['pass'] = false;
+			Email::send($CFG->form_email,$CFG->accounts_email,$email['title'],$CFG->form_email_from,false,$email['content'],$info);
+			return true;
+		}
+	}
+    
+
+	function registerAuthy($cell,$country_code) {
+		global $CFG;
+		
+		$cell = preg_replace("/[^0-9]/", "",$cell);
+		$country_code = preg_replace("/[^0-9]/", "",$country_code);
+		
+		if (!$CFG->session_active || User::$info['verified_authy'] == 'Y')
+			return false;
+		
+		$response = shell_exec("
```

Comment:
Why not use something safer like guzzle?

Reply:
This entire file wasn't actually changed by me in this case.

Reply:
Probably a line ending change

Reply:
Sorry, it's a terribly old PR, and I didn't realise until I posted. Anyway, as far as I can tell, this project's moved.